### DISCUSSION TITLE
Inconsistent "dative" dependency?

### DISCUSSION THREAD

**Discussion by justinbhopper** [explosion/spaCy#8445] [Help: Other Questions] [answered]
In the examples below, I would expect the word "**to**" to be a "**dative**" dependency in every case.  
However, instead "**to**" is getting marked as a prepositional modifier ("**prep**") with most verbs.  So far I have only found "give" and "send" result in dative.  

## How to reproduce the behaviour
```
import spacy
nlp = spacy.load("en_core_web_sm")

doc = nlp("I took the results to the mother.")
print(doc[4].dep_) # "prep"

doc = nlp("I gave the results to the mother.")
print(doc[4].dep_) # "dative"

doc = nlp("I sent the results to the mother.")
print(doc[4].dep_) # "dative"

doc = nlp("I emailed the results to the mother.")
print(doc[4].dep_) # "prep"
```

Is this expected?  Is there a linguistic rationale behind the different results, or is there just a limit on how well spaCy can detect the dative case?

## Your Environment
* Operating System: Windows 10
* Python Version Used: Python 3
* spaCy Version Used: v3.0
* Environment Information: Binder

**Comment by polm** [comment, marked as answer]
This is not exactly expected, but there are issues with the `dative` label that make it kind of unreliable, so it's also not something we can easily fix. If we did fix it the result would almost certainly be the removal of any detection of datives at all.

The `dative` label is not a standard label in Stanford NLP or Universal Dependencies; it's specific to [ClearNLP][], the software we use for converting our training corpus. We'd have to look at how the conversion is done to pin down the exact reasons for this, but it could be related to issues with the underlying annotations (since they probably weren't intended for this use) or sparsity of vocabulary (the training data is of an age and type of writing that "emailed" may not be present at all). 

[ClearNLP]: https://github.com/clir/clearnlp-guidelines

I think modern Universal Dependencies doesn't differentiate between these cases at all and they would both be marked as `case` ([docs](https://universaldependencies.org/u/dep/all.html#al-u-dep/case)). So if we were to revise things to come more in line with UD this distinction would probably just go away.

**Reply by justinbhopper** [reply]
I think considering removing `dative` entirely may be the right move, given all of the reasons you listed.  Having a dependency that appears unreliably also means that it might appear unexpectedly - where someone expected a `prep` and instead received a `dative`.

Perhaps it would be better to move it somewhere else so that it can be considered an _additional_ tag that may or may not be present?  I'm reminded of how the MorphAnalysis dictionary is designed, where certain keys are only sometimes present.

**Reply by adrianeboyd** [reply]
So many users rely on the English models with their current annotation schemes that we are unlikely to make this kind of change, but it would be pretty straightforward to write a custom component that collapses `dative` into `prep` or copies it into a `morph` feature as a postprocessing step, if that would be helpful for your task.

An additional English model with a UD v2 dependency parser is on our long-term to-do list, but the OntoNotes conversion is going to be tricky and time-consuming.

**Reply by justinbhopper** [reply]
@adrianeboyd Thanks for the clarification.  I definitely understand that would it be a problematic change to suddenly drop `dative`, since its likely to impact a ton of current implementations.  I agree it's easy on my part to simply collapse `dative` into `prep` now that I know to expect it, and to not count on `dative` being reported in all cases.

I think if I ever need to detect the dative case, I'll probably just try to detect it on my own using some basic happy-path rules (e.g. where "to" is a prep and one of the right siblings is an object).  Maybe I'll find that it's good-enough and reliable.
